### PR TITLE
Correctly handle code action of CodeAction type (fixes #566)

### LINKED ISSUE #566: Code actions of CodeActionLiteral type are not supported (used by eslint-server)
While experimenting with [eslint-server](https://github.com/Microsoft/vscode-eslint/tree/master/server), I've noticed that its code actions for fixing lint issues are not handled properly.

I found two issues:
 - handling CodeActions action list as an array of `Command` while protocol also allows array of `CodeAction`.
 - Passing `args` param to `workspace/executeCommand` instead of expected `arguments` param.

### PR DESCRIPTION
**Pull request by rchl**
Fixes two issues:
  - The Command's arguments passed to workspace/executeCommand should be
    passed through 'arguments' property rather than 'args'.
  - Fix handling of code action in case textDocument/codeAction [1] returns
    list of CodeAction's rather than Command's [2].

    Instead of having LspCodeActionsCommand duplicate logic of
    LspExecuteCommand, pass data from former to the latter to
    execute code action. Had to modify latter to not trigger a
    dialog box on success as it's not the best user experience in case
    when executing action from a popup menu.
  - Advertise support for workspace/executeCommand as we clearly support
    that. That wasn't necessary to fix anything but it felt right to do.

[1] https://microsoft.github.io/language-server-protocol/specification#textDocument_codeAction
[2] https://microsoft.github.io/language-server-protocol/specification#command

### CODE REVIEW

File: plugin/code_actions.py [hunk 1]
```diff
@@ -108,11 +108,18 @@ def show_popup_menu(self) -> None:
...
         if index > -1:
-            client = client_for_view(self.view)
-            if client:
-                client.send_request(
-                    Request.executeCommand(self.commands[index]),
-                    self.handle_command_response)
-
-    def handle_command_response(self, response):
-        pass
+            # TODO for CodeAction, handle potential WorkspaceEdit before Command.
```
Comment (tomv564) [inline plugin/code_actions.py]: I'd like to see this supported as part of the CodeAction implementation

File: plugin/code_actions.py [hunk 2]
```diff
@@ -108,11 +108,18 @@ def show_popup_menu(self) -> None:
...
+            if command:
+                command_name = command.get('command')
+                command_args = command.get('arguments', None)
+
+                args = {
+                    "command_name": command_name,
+                    "command_args": command_args,
+                    "modal_result": False,
+                }
+                self.view.run_command('lsp_execute', args)
```
Comment (tomv564) [inline plugin/code_actions.py]: Would prefer if the command execution didn't use `LspExecuteCommand`, that class is for letting the user directly run commands from the quick panel and not to be invoked from code.

It's fine if we want them to share functionality though!

File: plugin/execute_command.py [hunk 3]
```diff
@@ -14,27 +14,32 @@ class LspExecuteCommand(LspTextCommand):
     def __init__(self, view):
         super().__init__(view)
 
-    def run(self, edit, command_name=None, command_args=None) -> None:
+    def run(self, edit, command_name=None, command_args=None, modal_result=True) -> None:
```
Comment (tomv564) [inline plugin/execute_command.py]: I like the idea of letting the user specify `modal_result`, but this functionality was meant to be separate from the code action command execution. Can we move it to another PR?

File: plugin/execute_command.py [hunk 4]
```diff
@@ -14,27 +14,32 @@ class LspExecuteCommand(LspTextCommand):
...
         msg = "command {} failed. Reason: {}".format(command, error.get("message", "none provided by server :("))
         sublime.message_dialog(msg)
 
-    def _send_command(self, client: Client, command_name: str, command_args: 'Dict[str, Any]') -> None:
+    def _send_command(self, client: Client, command_name: str, command_args: 'Optional[List[Any]]',
+                      modal_result) -> None:
         request = {
             "command": command_name,
-            "args": command_args
+            "arguments": command_args
```
Comment (tomv564) [inline plugin/execute_command.py]: Sorry, I jumped the gun and fixed this in master as it was a clear bug!

File: plugin/core/sessions.py [hunk 5]
```diff
@@ -136,7 +136,8 @@ def get_initialize_params(project_path: str, config: ClientConfig):
             },
             "workspace": {
                 "applyEdit": True,
-                "didChangeConfiguration": {}
+                "didChangeConfiguration": {},
+                "executeCommand": {},
```
Comment (tomv564) [inline plugin/core/sessions.py]: Nice catch here as well!

**Comment by AmjadHD** [comment]
https://github.com/rchl/LSP/blob/1e048eb498a01a5e49a1617da3e18c662f59ded4/plugin/code_actions.py#L69
that failes if the response is None, which is the case of [lua ls](https://github.com/tomv564/LSP/issues/596).

**Comment by tomv564** [comment]
Tested with lsp-tsserver, hope it works for your use cases as well. Thanks again @rchl !

**Comment by ayoub-benali** [comment]
Thanks for the PR completion @tomv564
